### PR TITLE
[AIRFLOW-5143] Fix for potentially corrupted .jar

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
 
### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -226,6 +226,7 @@ RUN \
 if [[ "${APT_DEPS_IMAGE}" == "airflow-apt-deps-ci" ]]; then \
     echo "Downloading RAT from ${RAT_URL} to ${RAT_JAR}" \
     && curl -sL ${RAT_URL} > ${RAT_JAR} \
+    && jar -tf ${RAT_JAR} \
```
Comment (mik-laj) [inline Dockerfile]: Can you add " around the variable?I would feel calmer

**Review by mik-laj — APPROVED** [review]

**Review by BasPH — COMMENTED** [review]
Apache Rat also provides SHA256 checksums for verifying the integrity. WDYT of using those?

File: Dockerfile-checklicence [hunk 2]
```diff
@@ -46,7 +46,8 @@ ENV RAT_VERSION="${RAT_VERSION}" \
     RAT_URL="http://repo1.maven.org/maven2/org/apache/rat/apache-rat/${RAT_VERSION}/apache-rat-${RAT_VERSION}.jar"
```
Comment (Fokko) [inline Dockerfile-checklicence]: This has to be `https://`. I know that this isn't part of the PR, but we need to fix this.

File: Dockerfile-checklicence [hunk 3]
```diff
@@ -46,7 +46,8 @@ ENV RAT_VERSION="${RAT_VERSION}" \
...
 
 RUN echo "Downloading RAT from ${RAT_URL} to ${RAT_JAR}" \
-    && curl -sL ${RAT_URL} > ${RAT_JAR}
+    && curl -sL ${RAT_URL} > ${RAT_JAR}  \
+    && jar -tf ${RAT_JAR}
```
Comment (Fokko) [inline Dockerfile-checklicence]: Maybe we can do a checksum to verify the contents: https://www.apache.org/info/verification.html

**Review by Fokko — COMMENTED** [review]
Tnx for picking this up @potiuk

File: Dockerfile (hunk 1 above)
Comment (potiuk) [reply Dockerfile]: Added!

File: Dockerfile-checklicence (hunk 3 above)
Comment (potiuk) [reply Dockerfile-checklicence]: Good Idea. Added!

**Comment by potiuk** [comment]
I realised I already moved the licence check to separate Dockerfile - so I fixed it  now in the *RIGHT* place and removed it from the CI Dockerfile as it is not needed there !

File: Dockerfile-checklicence (hunk 2 above)
Comment (potiuk) [reply Dockerfile-checklicence]: Ah Right! fixed it as well ! Good catch @Fokko :)

**Review by kaxil — APPROVED** [review]
LGTM.

Thanks @potiuk

**Comment by potiuk** [comment]
One more pass. One of the tests failed and I realised that we've been building checklicence image needlessly in tests.

**Review by Fokko — APPROVED** [review]

**Pull request merged**